entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationService.kt

```diff
@@ -0,0 +1,84 @@
+package com.egm.stellio.entity.authorization
+
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.ADMIN_RIGHT
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.ADMIN_ROLES
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.CREATION_ROLES
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.READ_RIGHT
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.R_CAN_ADMIN
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.USER_PREFIX
+import com.egm.stellio.entity.authorization.AuthorizationService.Companion.WRITE_RIGHT
+import com.egm.stellio.entity.model.Relationship
+import com.egm.stellio.entity.repository.EntitySubjectNode
+import com.egm.stellio.entity.repository.Neo4jRepository
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import java.net.URI
+
+@Component
+@ConditionalOnProperty("application.authentication.enabled")
+class Neo4jAuthorizationService(
+    private val neo4jAuthorizationRepository: Neo4jAuthorizationRepository,
+    private val neo4jRepository: Neo4jRepository
+) : AuthorizationService {
+
+    override fun userIsAdmin(userId: String): Boolean = userIsOneOfGivenRoles(ADMIN_ROLES, userId)
+
+    override fun userIsCreator(userId: String): Boolean = userIsOneOfGivenRoles(CREATION_ROLES, userId)
+
+    private fun userIsOneOfGivenRoles(roles: Set<String>, userId: String): Boolean =
+        neo4jAuthorizationRepository.getUserRoles(USER_PREFIX + userId).intersect(roles).isNotEmpty()
+
+    override fun filterEntitiesUserHasReadRight(entitiesId: List<String>, userId: String): List<String> =
```

Comment:
If I understood correctly, if we want to filter the entities that a User can read, we get all the rights that the user has (and has thanks to his group) for this entities and then we filter the matching ones.

Can't we have, in the Neo4jAuthorizationRepository, functions that filters directly according to the provided Right ?

Reply:
Indeed this is a better approach, I'll try to do it this way.

Reply:
I tried to do this approach in 0e33a2
Could you give me a feedback ?

Reply:
Done